integration-libs/opf/base/root/model/opf.model.ts

```diff
@@ -0,0 +1,187 @@
+  javaScriptEnabled?: boolean;
+  language?: string;
+  screenHeight?: number;
+  screenWidth?: number;
+  userAgent?: string;
+  timeZoneOffset?: number;
+  ipAddress?: string;
+  originUrl?: string;
+}
+
+export interface SubmitRequest {
+  browserInfo?: PaymentBrowserInfo;
+  paymentMethod?: string;
+  encryptedToken?: string;
+  cartId?: string;
+  channel?: string;
+  additionalData?: Array<KeyValuePair>;
+}
+
+export interface SubmitInput {
+  additionalData: Array<KeyValuePair>;
+  paymentSessionId: string;
+  cartId: string;
+  callbackArray: [MerchantCallback, MerchantCallback, MerchantCallback];
+  returnPath?: string;
+  paymentMethod: PaymentMethod;
+  encryptedToken?: string;
+}
+
+export enum SubmitStatus {
+  REJECTED = 'REJECTED',
+  ACCEPTED = 'ACCEPTED',
+  PENDING = 'PENDING',
+  DELAYED = 'DELAYED',
+}
+export enum PaymentMethod {
+  APPLE_PAY = 'APPLE_PAY',
+  CREDIT_CARD = 'CREDIT_CARD',
+  GOOGLE_PAY = 'GOOGLE_PAY',
+}
+export interface SubmitResponse {
+  cartId: string;
+  status: SubmitStatus;
+  reasonCode: string;
+  paymentMethod: PaymentMethod;
+  authorizedAmount: number;
+  customFields: Array<KeyValuePair>;
+}
+
+export interface SubmitCompleteResponse {
+  cartId: string;
+  status: SubmitStatus;
+  reasonCode: number;
+  customFields: Array<KeyValuePair>;
+}
+
+export interface SubmitCompleteRequest {
+  paymentSessionId?: string;
+  additionalData?: Array<KeyValuePair>;
+  cartId?: string;
+}
+export interface SubmitCompleteInput {
+  additionalData: Array<KeyValuePair>;
+  paymentSessionId: string;
+  cartId: string;
+  callbackArray: [MerchantCallback, MerchantCallback, MerchantCallback];
+  returnPath?: string;
+}
+
+export interface AfterRedirectScriptResponse {
+  afterRedirectScript: OpfDynamicScript;
+}
+
+export interface OpfDynamicScript {
+  cssUrls?: OpfDynamicScriptResource[];
+  jsUrls?: OpfDynamicScriptResource[];
+  html?: string;
+}
+
+export interface OpfDynamicScriptResource {
+  url?: string;
+  sri?: string;
+  attributes?: KeyValuePair[];
+  type?: OpfDynamicScriptResourceType;
+}
+
+export enum OpfDynamicScriptResourceType {
+  SCRIPT = 'SCRIPT',
+  STYLES = 'STYLES',
+}
+
+export interface GlobalFunctionsInput {
+  paymentSessionId: string;
+  vcr?: ViewContainerRef;
+  paramsMap?: Array<KeyValuePair>;
+  domain: GlobalFunctionsDomain;
+}
+
+export enum TargetPage {
+  CHECKOUT_REVIEW = 'CHECKOUT_REVIEW',
+  RESULT = 'RESULT',
+}
+
+export type ErrorDialogOptions = {
+  confirmString?: string;
+  confirmKey?: string;
+  confirmReplacements?: any;
+  messageString?: string;
+  messageKey?: string;
+  messageReplacements?: any;
+};
+
+export const defaultErrorDialogOptions: ErrorDialogOptions = {
+  messageKey: 'opf.payment.errors.proceedPayment',
+  confirmKey: 'common.continue',
+};
+
+export enum OpfPage {
+  CHECKOUT_REVIEW_PAGE = 'opfCheckoutPaymentAndReview',
+  CONFIRMATION_PAGE = 'orderConfirmation',
+  RESULT_PAGE = 'paymentVerificationResult',
+  CART_PAGE = 'cart',
+}
```

Comment:
[MEDIUM]
1. For future easier reasoning and discoverability of the "this enum's keys VS semantic routes names", can we align identically the keys with the values?
  - `CONFIRMATION_PAGE = 'orderConfirmation'` -> rename key to `ORDER_CONFIRMATION`
  - `CHECKOUT_REVIEW_PAGE = 'opfCheckoutPaymentAndReview'` -> rename key to `OPF_CHECKOUT_PAYMENT_AND_REVIEW`
  - etc.

2. What about `OpfPage` vs `OpfSemanticRoutes`?
3. Do we really need the suffix `_PAGE` in each key? i.e. what about `OpfSemanticRoutes.CART`, `OpfSemanticRoutes.ORDER_CONFIRMATION`, `OpfSemanticRoutes.OPF_CHECKOUT_PAYMENT_AND_REVIEW